service/history/mutableStateBuilder.go

```diff
@@ -1070,6 +1071,7 @@ func (e *mutableStateBuilder) AddContinueAsNewEvent(decisionCompletedEventID int
 		InitiatedID:          initiatedID,
 		TaskList:             newStateBuilder.executionInfo.TaskList,
 		WorkflowTypeName:     newStateBuilder.executionInfo.WorkflowTypeName,
+		WorkflowTimeout:      newStateBuilder.executionInfo.WorkflowTimeout,
```

Comment:
Can you have a test to make sure WorkflowTimeout is valid after new run of execution is started.

Reply:
Will work on adding an integration test for this.